### ISSUE TITLE
Enable not working on SG300 

### ISSUE THREAD

**Issue by ckhordiasma** [ansible-collections/community.ciscosmb#73]
##### SUMMARY
I'm trying to run commands from a privilege level 7 account that can elevate to 15 using enable. What I am finding is that the commands are not running, which seems to indicate that the elevation to privilege 15 is not happening. When I try to do the same reproduction steps with a privilege level 15 account, the issue does not appear.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
community.ciscosmb.command

##### ANSIBLE VERSION

```paste below
ansible [core 2.15.9]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.18 (main, Feb 13 2024, 10:54:04) [GCC 12.2.0] (/usr/local/bin/python)
  jinja version = 3.1.3
  libyaml = True
```

##### COLLECTION VERSION

```paste below
# /usr/local/lib/python3.9/site-packages/ansible_collections
Collection         Version
------------------ -------
community.ciscosmb 1.0.7  
```

##### CONFIGURATION
inventory.yml config for this switch:
```
switch1:
      ansible_host: xx.xx.xx.xx
      ansible_network_os: community.ciscosmb.ciscosmb
      ansible_ssh_private_key_file: ./cisco-privkey
      ansible_ssh_user: username
      ansible_become_method: "enable"
      ansible_become_password: "******"
      ansible_ssh_port: xxxx
      ansible_host_key_checking: false
      new_user_password: "example_password"
```

##### OS / ENVIRONMENT

```
NAME: "1"   DESCR: "SG300-20 20-Port Gigabit Managed Switch"
SW version    1.4.11.5 ( date  08-Apr-2020 time  13:49:34 )
Boot version    1.3.5.06 ( date  21-Jul-2013 time  15:12:10 )
HW version    V04
```

##### STEPS TO REPRODUCE

playbook task that doesn't work:

```yaml
    - name: Set up ansible user account
      become: yes
      community.ciscosmb.command:
        commands:
          - config t
          - "username ansible password {{new_user_password}} privilege 15"
          - ip ssh pubkey-auth auto-login
          - exit
```

##### EXPECTED RESULTS
All commands work as expected

##### ACTUAL RESULTS
password and subsequent commands are not recognized, indicating that enable didn't activate properly

```paste below
redirecting (type: connection) ansible.builtin.network_cli to ansible.netcommon.network_cli
redirecting (type: become) ansible.builtin.enable to ansible.netcommon.enable
<192.168.42.104> ESTABLISH LOCAL CONNECTION FOR USER: root
<192.168.42.104> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-local-1pwj0mr4g `"&& mkdir "` echo /root/.ansible/tmp/ansible-local-1pwj0mr4g/ansible-tmp-1710593382.4371493-9-223457125418386 `" && echo ansible-tmp-1710593382.4371493-9-223457125418386="` echo /root/.ansible/tmp/ansible-local-1pwj0mr4g/ansible-tmp-1710593382.4371493-9-223457125418386 `" ) && sleep 0'
Using module file /usr/local/lib/python3.9/site-packages/ansible_collections/community/ciscosmb/plugins/modules/command.py
<192.168.42.104> PUT /root/.ansible/tmp/ansible-local-1pwj0mr4g/tmp7_39ad_r TO /root/.ansible/tmp/ansible-local-1pwj0mr4g/ansible-tmp-1710593382.4371493-9-223457125418386/AnsiballZ_command.py
<192.168.42.104> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-local-1pwj0mr4g/ansible-tmp-1710593382.4371493-9-223457125418386/ /root/.ansible/tmp/ansible-local-1pwj0mr4g/ansible-tmp-1710593382.4371493-9-223457125418386/AnsiballZ_command.py && sleep 0'
<192.168.42.104> EXEC /bin/sh -c '/usr/local/bin/python /root/.ansible/tmp/ansible-local-1pwj0mr4g/ansible-tmp-1710593382.4371493-9-223457125418386/AnsiballZ_command.py && sleep 0'
<192.168.42.104> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-local-1pwj0mr4g/ansible-tmp-1710593382.4371493-9-223457125418386/ > /dev/null 2>&1 && sleep 0'
ok: [switch1_chkodama] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "commands": [
                "config t",
                "username ansible password example_password privilege 15",
                "ip ssh pubkey-auth auto-login",
                "exit"
            ],
            "interval": 1,
            "match": "all",
            "retries": 10,
            "wait_for": null
        }
    },
    "stdout": [
        "switch(config)#",
        "% Unrecognized command\nswitch(config)#",
        "% Unrecognized command\nswitch(config)#",
        "switch#"
    ],
    "stdout_lines": [
        [
            "switch(config)#"
        ],
        [
            "% Unrecognized command",
            "switch(config)#"
        ],
        [
            "% Unrecognized command",
            "switch(config)#"
        ],
        [
            "switch#"
        ]
    ]
}
```

**Comment by ckhordiasma** [comment]
From looking at the code, I think it's because of this

**Comment by ckhordiasma** [comment]
alright... found a workaround. If I add the following task before my non-working task

```
    - name: disable
      become: no
      community.ciscosmb.command:
        commands:
          - disable
```

AND ensure that `ansible_become` is unset or set to no in the inventory file,

then my original task works as expected.

**Comment by qaxi** [comment]
Hi

my understanding to _become: yes_ is this: immediately after login it (on cisco sw) sends  _enable_ command. 

And the priv. 7 account looks like "edgy" ... but legit.

I should check privilege more exactly.

Can you  test this and send me output?

```
    - name: Set up ansible user account
      become: yes
      community.ciscosmb.command:
        commands:
          - sh privilege
          - enable
          - sh privilege
          - config t
          - "username ansible password {{new_user_password}} privilege 15"
          - ip ssh pubkey-auth auto-login
          - exit
```

And just in case test this
```
    - name: Set up ansible user account
      become: yes
      community.ciscosmb.command:
        commands:
          - enable
          - config t
          - "username ansible password {{new_user_password}} privilege 15"
          - ip ssh pubkey-auth auto-login
          - exit
```

**Comment by qaxi** [comment]
Never mind, I just checked the code and the leading test is redundant.

I will delete it and release ASAP.

Thank you for help.

**Comment by qaxi** [comment]
Resolved by PR #74

**Comment by ckhordiasma** [comment]
thank you!
